js/data/bucket/symbol_bucket.js

```diff
@@ -333,6 +340,45 @@ SymbolBucket.prototype.addFeature = function(lines, shapedText, shapedIcon, feat
     }
 };
 
+SymbolBucket.prototype.findPolygonAnchors = function(line, lines, polygons) {
+
+    if (line.length === 0) {
+        return [];
+    } else if (line.length < 3) {
+        return [ new Anchor(line[0].x, line[0].y, 0) ];
+    }
+
+    var anchors = null;
+    // Work out what line is...
+    var polygon = null;
+    var isPolygon = false;
+    var isHole = false;
+    for (var i = 0; !isPolygon && !isHole && i < polygons.length; i++) {
+        polygon = polygons[i];
+        isPolygon = line === polygon[0];
+        if (!isPolygon) {
+            for (var j = 1; !isHole && j < polygon.length; j++) {
+                isHole = line === polygon[i];
+            }
+        }
+    }
+
+    if (isPolygon) {
+        var poi = poleOfInaccessibility(polygon, 1);
```

Comment:
Do you think we need the `1` precision?  It could probably be bumped to something like 16 (which is just 2 pixels).